microprofile/health-smallrye/src/main/java/org/wildfly/extension/microprofile/health/deployment/CDIExtension.java

```diff
@@ -67,24 +67,28 @@ private void afterDeploymentValidation(@Observes final AfterDeploymentValidation
         addHealthChecks(Readiness.Literal.INSTANCE, reporter::addReadinessCheck, readinessChecks);
         addHealthChecks(Startup.Literal.INSTANCE, reporter::addStartupCheck, startupChecks);
         reporter.setUserChecksProcessed(true);
+
+        final Config config = ConfigProvider.getConfig(module.getClassLoader());
+        final boolean disableDefaultprocedure = config.getOptionalValue(MP_HEALTH_DISABLE_DEFAULT_PROCEDURES, Boolean.class).orElse(false);
         if (readinessChecks.isEmpty()) {
-            Config config = ConfigProvider.getConfig(module.getClassLoader());
-            boolean disableDefaultprocedure = config.getOptionalValue(MP_HEALTH_DISABLE_DEFAULT_PROCEDURES, Boolean.class).orElse(false);
             if (!disableDefaultprocedure) {
                 // no readiness probe are present in the deployment. register a readiness check so that the deployment is considered ready
                 defaultReadinessCheck = new DefaultReadinessHealthCheck(module.getName());
                 reporter.addReadinessCheck(defaultReadinessCheck, module.getClassLoader());
             }
         }
         if (startupChecks.isEmpty()) {
-            Config config = ConfigProvider.getConfig(module.getClassLoader());
-            boolean disableDefaultprocedure = config.getOptionalValue(MP_HEALTH_DISABLE_DEFAULT_PROCEDURES, Boolean.class).orElse(false);
             if (!disableDefaultprocedure) {
                 // no startup probes are present in the deployment. register a startup check so that the deployment is considered started
                 defaultStartupCheck = new DefaultStartupHealthCheck(module.getName());
                 reporter.addStartupCheck(defaultStartupCheck, module.getClassLoader());
             }
         }
+        // https://issues.redhat.com/browse/WFLY-19147 - let the per-application `mp.health.disable-default-procedures`
+        // configuration setting affect the Health checks response global server configuration
+        if (disableDefaultprocedure) {
+            reporter.deactivateDefaultProcedures(module.getName());
```

Comment:
module.getName() isn't the user-understandable deployment name. We don't have the info here to create an 'ideal' user-understandable deployment name in all cases, but removing any preceding ServiceModuleLoader.MODULE_PREFIX will come pretty close.